test/cql-pytest/conftest.py

```diff
@@ -28,6 +28,10 @@ def pytest_addoption(parser):
         help='CQL server port to connect to')
     parser.addoption('--ssl', action='store_true',
         help='Connect to CQL via an encrypted TLSv1.2 connection')
+    parser.addoption('--scylla-path', action='store',
+        help='Path to the scylla excutable the tests are running against')
+    parser.addoption('--workdir', action='store',
+        help='Path to the workdir of the scylla instance the tests are running against')
```

Comment:
Well we have `data_file_directories` in `system.config`, which is all we really need from `workdir`, so that is sorted.

Reply:
I guess I can add scylla's argv0 to one of the system tables. I agree that the test would be much cleaner without these extra parameters.

Reply:
By the way, note that even if you fetch workdir and scylla_path via the CQL protocol, there is still a possibility that these files are not available (e.g., Scylla is running on that path - but in a different machine!!) so you would need to check that possibility too and skip the test. It's not trivial to test, by the way, you can't just check if the directory exists locally - it may exist but it's still a different machine :-) The most foolproof way to test is to create a random-named keyspace in CQL and check if a directory with that name appears in workdir. You don't even need to create another keyspace - you can use the existing test_keyspace fixture which anyway needs to create a random keyspace, and it will be used by most of the tests anyway.

Reply:
A thought: Instead of the path, save the process ID. You can then use /proc to find the process's full pathname. As before, this process ID should be tested (by sending it signal 0), because it's possible it's a process ID on a different machine ;-)

Reply:
That might work. But this starts to sound like a project. Which I don't mind to to do but I don't want all the JSON upgrade code to bitrot due to an improvement to the tests. So would you be willing to merge this with the skip solution with me attempting the nicer one as a follow-up?

Reply:
Yes, no problem.
So now I'm just waiting for a new version with just the added if and skip()?